zebedee/client_test.go

```diff
@@ -361,22 +383,86 @@ func TestClient_HealthChecker(t *testing.T) {
+	body := httpmocks.NewReadCloserMock(documentContent, nil)
+
+	Convey("given a 200 response", t, func() {
+		response := httpmocks.NewResponseMock(body, http.StatusOK)
+		httpClient := newMockHTTPClient(response, nil)
+		zebedeeClient := newZebedeeClient(httpClient)
+
+		Convey("when zebedeeClient.GetPublised is called", func() {
+			testContent, err := zebedeeClient.GetPublishedData(ctx, testURIString)
+
+			Convey("then the expected content is returned", func() {
+				So(err, ShouldBeNil)
+				So(testContent, ShouldNotBeNil)
+				So(testContent, ShouldResemble, documentContent)
+			})
+
+			Convey("and client.Do should be called once with the expected parameters", func() {
+				doCalls := httpClient.DoCalls()
+				So(doCalls, ShouldHaveLength, 1)
+				So(doCalls[0].Req.URL.Path, ShouldEqual, path)
+				p := doCalls[0].Req.FormValue("uri")
+				So(p, ShouldEqual, testURIString)
+			})
+		})
+	})
+
+	Convey("given a 500 response", t, func() {
+		response := httpmocks.NewResponseMock(body, http.StatusInternalServerError)
+		httpClient := newMockHTTPClient(response, nil)
+		zebedeeClient := newZebedeeClient(httpClient)
+
+		Convey("when zebedeeClient.Checker is called", func() {
+			testContent, err := zebedeeClient.GetPublishedData(ctx, testURIString)
+
+			Convey("then the expected check is returned", func() {
+				// So(err.Error(), ShouldResemble, errors.Errorf(http.StatusInternalServerError).Error())
+				// So(err.Error(), ShouldResemble, errors.Errorf("invalid response: 500 from image api: http://localhost:8080/images, body: wrong!").Error())
+				So(err, ShouldNotBeNil)
+				So(testContent, ShouldBeNil)
+				So(err, ShouldHaveSameTypeAs, ErrInvalidZebedeeResponse{})
+				So(err.Error(), ShouldEqual, "invalid response from zebedee - should be 2.x.x or 3.x.x, got: 500, path: /publisheddata")
+
+			})
+
+			Convey("and client.Do should be called once with the expected parameters", func() {
+				doCalls := httpClient.DoCalls()
+				So(doCalls, ShouldHaveLength, 1)
+				So(doCalls[0].Req.URL.Path, ShouldEqual, path)
+			})
+		})
+	})
+	Convey("given a 404 response", t, func() {
+		response := httpmocks.NewResponseMock(body, http.StatusNotFound)
+		httpClient := newMockHTTPClient(response, nil)
+		zebedeeClient := newZebedeeClient(httpClient)
+
+		Convey("when zebedeeClient.Checker is called", func() {
+			testContent, err := zebedeeClient.GetPublishedData(ctx, testURIString)
+
+			Convey("then the expected check is returned", func() {
+				// So(err.Error(), ShouldResemble, errors.Errorf(http.StatusInternalServerError).Error())
+				// So(err.Error(), ShouldResemble, errors.Errorf("invalid response: 500 from image api: http://localhost:8080/images, body: wrong!").Error())
+				So(err, ShouldNotBeNil)
+				So(testContent, ShouldBeNil)
+				So(err, ShouldHaveSameTypeAs, ErrInvalidZebedeeResponse{})
+				So(err.Error(), ShouldEqual, "invalid response from zebedee - should be 2.x.x or 3.x.x, got: 404, path: /publisheddata")
```

Comment:
I'm not sure this is a meaningful error for the client to be returning when a document cannot be found. I suggest we do some work on this to return something more effective (like say "page not found :/uri/of/the/page" or something like that).